compiler/transforms/set_memory_layout.py

```diff
@@ -121,6 +126,8 @@ def match_and_rewrite(self, linalg_op: linalg.Generic, rewriter: PatternRewriter
         else:
             library_call = linalg_op.library_call.data
 
+        assert self.gemm_layout in ("cyclic", "banked")
+
```

Comment:
Would it make sense to make this a StrEnum instead?

Reply:
ok!